src/rust/ide/src/controller/ide/desktop.rs

```diff
@@ -97,7 +97,7 @@ impl ManagingProjectAPI for Handle {
             let with_suffix           = (1..).map(|i| format!("{}_{}", UNNAMED_PROJECT_NAME, i));
             let mut candidates        = std::iter::once(without_suffix).chain(with_suffix);
             // The iterator have no end, so we can safely unwrap.
-            let name    = candidates.find(|c| names.contains(c)).unwrap();
+            let name    = candidates.find(|c| !names.contains(c)).unwrap();
```

Comment:
This change doesn’t seem to be related to cursor, is it? If not, what it is about?

Reply:
It's related to the second part of PR titile: `Bug in Creating New Projects`

Reply:
Yes, but the bug is not described anywhere. I'd like to understand what it was and how this change fixes it. Any comment about it would be helpful. Also, is this something that should be added to the changelog?